### PR TITLE
Tomcat install script

### PR DESCRIPTION
**Pull request by jasonsouza**

### CODE REVIEW

File: packages/tomcat.sh [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+#
+# change the following environment variables to your project configuration as needed,
+# otherwise the defaults below will be used.
+# * TOMCAT_VERSION
+#
+# Include the following command your builds:
+# \curl -sSL https://raw.githubusercontent.com/codeship/scripts/master/packages/tomcat.sh | bash -s
+
+TOMCAT_VERSION=${TOMCAT_VERSION:="8.5.9"}
+TOMCAT_DIR=${TOMCAT_DIR="$HOME/tocat"}
```
Comment (mlocher) [inline packages/tomcat.sh]: typo with the `TOMCAT_DIR`

File: packages/tomcat.sh [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+TOMCAT_WAIT_TIME=${TOMCAT_WAIT_TIME:="10"}
+
+set -e
+CACHED_DOWNLOAD="${HOME}/cache/apache-tomcat-${TOMCAT_VERSION}.tar.gz"
+
+mkdir -p "${TOMCAT_DIR}"
+wget --continue --output-document "${CACHED_DOWNLOAD}" "http://www-us.apache.org/dist/tomcat/tomcat-8/v${TOMCAT_VERSION}/bin/apache-tomcat-${TOMCAT_VERSION}.tar.gz"
+tar -xaf "${CACHED_DOWNLOAD}" --directory "${TOMCAT_DIR}"
```
Comment (mlocher) [inline packages/tomcat.sh]: Could you add `--strip-components=1`, which will extract the contents directly to `TOMCAT_DIR` and not to a subdirectory called `apache-tomcat-${TOMCAT_VERSION}`

File: packages/tomcat.sh [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+# Make sure to use the exact parameters you want for Tomcat and give it enough sleep time to properly start up
+bash ${TOMCAT_DIR}/bash apache-tomcat-${TOMCAT_VERSION}/bin/startup.sh
```
Comment (mlocher) [inline packages/tomcat.sh]: With the changes to the `tar` command above, we can simplify this command.

File: packages/tomcat.sh [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+sleep ${TOMCAT_SLEEP_TIME}
+cd -
```
Comment (mlocher) [inline packages/tomcat.sh]: As we're not `cd`'ing to a different directory, we don't need to switch back and can remove this command.

**Review by mlocher — CHANGES_REQUESTED** [review]
Great to get this added, I left a couple comments to streamline the script a bit.

**Comment by mlocher** [comment]
Please also add a test installation to the `test.sh` script, so we can make sure the script works as expected.

File: packages/tomcat.sh [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+TOMCAT_VERSION=${TOMCAT_VERSION:="8.5.9"}
+TOMCAT_DIR=${TOMCAT_DIR="$HOME/tocat"}
+TOMCAT_WAIT_TIME=${TOMCAT_WAIT_TIME:="10"}
+
+set -e
+CACHED_DOWNLOAD="${HOME}/cache/apache-tomcat-${TOMCAT_VERSION}.tar.gz"
+
+mkdir -p "${TOMCAT_DIR}"
+wget --continue --output-document "${CACHED_DOWNLOAD}" "http://www-us.apache.org/dist/tomcat/tomcat-8/v${TOMCAT_VERSION}/bin/apache-tomcat-${TOMCAT_VERSION}.tar.gz"
```
Comment (mlocher) [inline packages/tomcat.sh]: I just noticed, that the URL includes the major version on the directory level as well. An example of how to parse the major version from a variable is available at https://github.com/codeship/scripts/blob/master/packages/elasticsearch.sh#L19

File: packages/tomcat.sh (hunk 4 above)
Comment (jasonsouza) [reply packages/tomcat.sh]: i screwed this part up when before i push the last commit. Just changed it back to what I originally had.

File: packages/tomcat.sh [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+mkdir -p "${TOMCAT_DIR}"
+wget --continue --output-document "${CACHED_DOWNLOAD}" "http://www-us.apache.org/dist/tomcat/tomcat-${TOMCAT_VERSION:0:1}/v${TOMCAT_VERSION}/bin/apache-tomcat-${TOMCAT_VERSION}.tar.gz"
+tar -xaf "${CACHED_DOWNLOAD}" --strip-components=1 --directory "${TOMCAT_DIR}"
+
+# Make sure to use the exact parameters you want for Tomcat and give it enough sleep time to properly start up
+bash ${TOMCAT_DIR}/bin/startup.sh
+sleep ${TOMCAT_WAIT_TIME}
+
+# check if tomcat successfully installs
+bash tomcat/bin/version.sh | grep "Apache Tomcat/${TOMCAT_VERSION}"
```
Comment (VanTanev) [inline packages/tomcat.sh]: @BoVice Shouldn't this be `bash ${TOMCAT_DIR}/bin/version.sh | grep "Apache Tomcat/${TOMCAT_VERSION}"` ?

Comment (jasonsouza) [reply packages/tomcat.sh]: Thanks for the catch! @VanTanev

**Comment by VanTanev** [comment]
@BoVice while this is not yet merged:

I found myself needing to start tomcat with catalina instead of the normal server startup, ie:
 - Setup tomcat (don't start)
 - Add webapps to `$TOMCAT_DIR/webapps/X`
 - Start with `${TOMCAT_DIR}/bin/catalina.sh run`

To that effect, it might be useful to have an env var like 
```bash
TOMCAT_AUTOSTART=${TOMCAT_AUTOSTART:="1"}
```

and do 
```bash
if [ "$TOMCAT_AUTOSTART" -eq "1" ]; then
  # Make sure to use the exact parameters you want for Tomcat and give it enough sleep time to properly start up
  bash ${TOMCAT_DIR}/bin/startup.sh
  sleep ${TOMCAT_WAIT_TIME}
fi
```

so that users may start the server when they need it.

**Comment by jasonsouza** [comment]
Thanks for another great suggestion @VanTanev. Added to the script 😄 Sorry for the delay in revisiting this script.

**Review by mlocher — APPROVED** [review]
:shipit: if the build is green

**Pull request merged**